src/main/java/org/terasology/launcher/repositories/GithubRepositoryAdapter.java

```diff
@@ -40,7 +41,14 @@ public GithubRepositoryAdapter(final OkHttpClient httpClient) {
                     .withConnector(new OkHttpConnector(httpClient))
                     .build();
             logger.debug("Github rate limit: {}", github.getRateLimit());
+        } catch (HttpException e) {
+            if (e.getResponseCode() == -1) {
+                // no internet connection, do nothing
+            } else {
+                e.printStackTrace();
+            }
```

Comment:
Hm, PMD says this is bad, and instead we should do
```suggestion
            if (e.getResponseCode() != -1) {
               // -1 would indicate no internet connection, in which case the exception is expected
               // and should just be ignored. In all other cases, this is indeed exceptional behavior
               // and we print the stacktrace.
               e.printStackTrace();
            }
```

I personally prefer the variant with the empty if branch, because I think the reasoning for the control flow is more comprehensible.

Reply:
I believe that your variant is fine. I'd say we can add a nopmd comment: https://pmd.github.io/pmd/pmd_userdocs_suppressing_warnings.html#nopmd-comment
The example they give even is an empty if statement 😁

Reply:
Thanks for adding the `//NOPMD` comments 👍